metadata-ingestion/docs/sources/iceberg/iceberg.md

```diff
@@ -0,0 +1,23 @@
+### Concept Mapping
+
+<!-- This should be a manual mapping of concepts from the source to the DataHub Metadata Model -->
+<!-- Authors should provide as much context as possible about how this mapping was generated, including assumptions made, known shortcuts, & any other caveats -->
+
+This ingestion source maps the following Source System Concepts to DataHub Concepts:
+
+<!-- Remove all unnecessary/irrevant DataHub Concepts -->
+
+| Source Concept | DataHub Concept | Notes |
+| -- | -- | -- |
+| `iceberg` | [Data Platform](docs/generated/metamodel/entities/dataPlatform.md) | |
+| Table | [Dataset](docs/generated/metamodel/entities/dataset.md) | Each Iceberg table maps to a Dataset named using the parent folders.  If a table is stored under `my/namespace/table`, the dataset name will be `my.namespace.table`.  If a [Platform Instance](https://datahubproject.io/docs/platform-instances/) is configured, it will be used as a prefix: `<platform_instance>.my.namespace.table`. |
```

Comment:
For the Hive catalog, the paths in storage are slightly different.

1. HiveCatalog based tables only support one level of namespace. Most other catalogs support the pathing used here though.
2. The path _does_ have an extra string in it, a `.db` constant, so the path winds up being `${warehouseLocation}/${namespace}.db/${tableName}`. (note the `.db` after the namespace).

Reply:
This code currently only supports tables created with a HadoopCatalog.  DataHub already has a [Hive source](https://datahubproject.io/docs/generated/ingestion/sources/hive), but I have never used it.  Once the new Iceberg Python API is released, which will include an abstraction for catalogs, this code will be refactored and I suppose that it will support tables created with any catalog types.  Then we'll have to look at how to create/normalize paths...  Does that make sense @kbendick ?